src/main/java/org/elasticsearch/cluster/metadata/MappingMetaData.java

```diff
@@ -160,11 +161,27 @@ public int hashCode() {
 
     public static class Timestamp {
 
-        public static String parseStringTimestamp(String timestampAsString, FormatDateTimeFormatter dateTimeFormatter) throws TimestampParsingException {
-            try {
-                return Long.toString(dateTimeFormatter.parser().parseMillis(timestampAsString));
-            } catch (RuntimeException e) {
-                throw new TimestampParsingException(timestampAsString, e);
+        private static final FormatDateTimeFormatter EPOCH_MILLIS_PARSER = Joda.forPattern("epoch_millis");
+
+        public static String parseStringTimestamp(String timestampAsString, FormatDateTimeFormatter dateTimeFormatter,
+                                                  Version version) throws TimestampParsingException {
+            // no need for unix timestamp parsing in 2.x
+            if (version.onOrAfter(Version.V_2_0_0)) {
+                try {
+                    return Long.toString(dateTimeFormatter.parser().parseMillis(timestampAsString));
```

Comment:
I think this could be simplified?
```
if (version.before(Version.V_2_0_0)) {
    dateTimeFormatter = EPOCH_MILLIS_PARSER;
}
... try/catch as it existed before
```